src/main/java/tennisclub/dao/CourtDaoImpl.java

```diff
@@ -20,10 +21,32 @@ public void create(Court court) {
     }
 
     @Override
-    public List<Court> findByLocation(String location) {
+    @Transactional
```

Comment:
Not sure whether DAO methods should be transactional as they will be probably a part of a transaction called by an upper level

Reply:
I believe on the lecture it was mentioned that the @Transactional annotation can be either marked before the start of the method or before the definition the class _(in this case, the Transactional will be then applied to every method)_, but it wasn't mentioned where the transaction should be started _(whether on the upper level or in the DAO implementations)_ - I am suggesting doing it according to the lecture (so the way Pavel did it is correct), so people who aren't very well versed in this field _(for example reviewers and so on)_ would have it easier 😊

Reply:
That sounds good!